wit/bindgen/generator.go

```diff
@@ -881,6 +899,39 @@ func (g *generator) variantRep(file *gen.File, dir wit.Direction, t *wit.TypeDef
 	stringio.Write(&b, "return ", stringsName, "[v.Tag()]\n")
 	b.WriteString("}\n\n")
 
+	file.Import("encoding/json")
+	b.WriteString(formatDocComments("MarshalJSON implements [json.Marshaler].", true))
+	stringio.Write(&b, "func (v ", goName, ") MarshalJSON() ([]byte, error) {\n")
+	stringio.Write(&b, "ret := make(map[string]any)\n")
```

Comment:
I think there might be a better way to do this. 

Possibly by putting MarshalJSON and UnmarshalJSON in package cm, and generate only the minimal mapping from case name to case type.